include/crocoddyl/core/numdiff/diff-action.hxx

```diff
@@ -18,7 +18,7 @@ DifferentialActionModelNumDiffTpl<Scalar>::DifferentialActionModelNumDiffTpl(boo
     : Base(model->get_state(), model->get_nu(), model->get_nr()), model_(model) {
   with_gauss_approx_ = with_gauss_approx;
   disturbance_ = std::sqrt(2.0 * std::numeric_limits<Scalar>::epsilon());
-  assert_pretty((!with_gauss_approx_ || nr_ > 1), "No Gauss approximation possible with nr = 1");
+  if (with_gauss_approx_ && nr_ == 1) throw_pretty("No Gauss approximation possible with nr = 1");
```

Comment:
@wxmerkt I think you made this change, but I somehow lose its track.
Could you explain me the reasoning behave the change from `assert_pretty` to `throw_pretty`?

Reply:
This is the initialiser (not a frequently called function): We do definitely want to check arguments during initialisation - this can otherwise cause a runtime issue.

Reply:
Thank you for the explanation.